internal/services/automation/automation_runtime_environment_resource.go

```diff
@@ -0,0 +1,306 @@
+	}
+}
+
+func (m AutomationRuntimeEnvironmentResource) Arguments() map[string]*pluginsdk.Schema {
+	return map[string]*pluginsdk.Schema{
+		"name": {
+			Type:         pluginsdk.TypeString,
+			Required:     true,
+			ForceNew:     true,
+			ValidateFunc: validation.StringIsNotEmpty,
+		},
+
+		"resource_group_name": commonschema.ResourceGroupName(),
+
+		"location": commonschema.Location(),
+
+		"automation_account_name": {
+			Type:         pluginsdk.TypeString,
+			Required:     true,
+			ForceNew:     true,
+			ValidateFunc: validate.AutomationAccount(),
+		},
+
+		"runtime_language": {
+			Type:     pluginsdk.TypeString,
+			Required: true,
+			ForceNew: true,
+			ValidateFunc: validation.StringInSlice([]string{
+				string("Python"),
+				string("PowerShell"),
+			}, false),
+		},
+
+		"runtime_version": {
+			Type:         pluginsdk.TypeString,
+			Required:     true,
+			ForceNew:     true,
+			ValidateFunc: validation.StringIsNotEmpty,
+		},
+
+		"description": {
+			Type:         pluginsdk.TypeString,
+			Optional:     true,
+			ValidateFunc: validation.StringIsNotEmpty,
+		},
+
+		"runtime_default_packages": {
+			Type:     pluginsdk.TypeMap,
+			Optional: true,
+			Elem:     &pluginsdk.Schema{Type: pluginsdk.TypeString},
+		},
+
+		"tags": commonschema.Tags(),
+	}
+}
+
+func (m AutomationRuntimeEnvironmentResource) Attributes() map[string]*pluginsdk.Schema {
+	return map[string]*pluginsdk.Schema{}
+}
+
+func (m AutomationRuntimeEnvironmentResource) ModelObject() interface{} {
+	return &AutomationRuntimeEnvironmentResourceModel{}
+}
+
+func (m AutomationRuntimeEnvironmentResource) ResourceType() string {
+	return "azurerm_automation_runtime_environment"
+}
+
+func (m AutomationRuntimeEnvironmentResource) Create() sdk.ResourceFunc {
+	return sdk.ResourceFunc{
+		Timeout: 10 * time.Minute,
+		Func: func(ctx context.Context, meta sdk.ResourceMetaData) error {
+			client := meta.Client.Automation.RuntimeEnvironment
+
+			var model AutomationRuntimeEnvironmentResourceModel
+			if err := meta.Decode(&model); err != nil {
+				return fmt.Errorf("decoding: %+v", err)
+			}
+
+			subscriptionID := meta.Client.Account.SubscriptionId
+			id := runtimeenvironment.NewRuntimeEnvironmentID(subscriptionID, model.ResourceGroupName, model.AutomationAccountName, model.Name)
+
+			existing, err := client.Get(ctx, id)
+			if err != nil && !response.WasNotFound(existing.HttpResponse) {
+				return fmt.Errorf("checking for presence of existing %s: %+v", id, err)
+			}
+
+			if !response.WasNotFound(existing.HttpResponse) {
+				return meta.ResourceRequiresImport(m.ResourceType(), id)
+			}
+
+			req := runtimeenvironment.RuntimeEnvironment{}
+			req.Properties = &runtimeenvironment.RuntimeEnvironmentProperties{
+				Runtime: &runtimeenvironment.RuntimeProperties{
+					Language: pointer.FromString(model.RuntimeLanguage),
+					Version:  pointer.FromString(model.RuntimeVersion),
+				},
+			}
+			req.Location = azure.NormalizeLocation(model.Location)
```

Comment:
A couple comments on this section,

- When building the payload/request object in the create, we would instantiate it in one go (for properties that are not conditionally included)
- `azure.NormalizeLocation` will be removed in the future in favour of `location.Normalize`, for new code additions, we should use the latter
- Could you replace `pointer.FromString` with the generic `pointer.To`?

```suggestion
			req := runtimeenvironment.RuntimeEnvironment{
				Location: location.Normalize(model.Location),
				Properties: &runtimeenvironment.RuntimeEnvironmentProperties{
					Runtime: &runtimeenvironment.RuntimeProperties{
						Language: pointer.To(model.RuntimeLanguage),
						Version:  pointer.To(model.RuntimeVersion),
					},
				},
			}
```